### ISSUE TITLE
Possible bug in fset:compare

### ISSUE THREAD

**Issue by khinsen** [slburson/fset#56]
In using FSet to work with system dependency graphs, I noticed some surprising behavior which I managed to cook down to the following example:

```
(let* ((system (asdf:find-system "babel"))
       (deps (mapcar #'asdf:find-system (asdf:system-depends-on system))))
  (fset::compare
   (cons system (first deps))
   (cons system (second deps))))
```
This returns `:equal`, although the two cons pairs have different cdrs.

Looking at `compare-lists-lexicographically`, this problem manifests itself for all dotted lists that (1) differ only in the last cdr and (2) have a last cdr that is of type `atom` but has no order relation defined. In my case, that's true for ASDF system objects.

What happens in this case is that the result form of the `do` returns `default` (which remains `:equal` all the time) even though `comp` (the comparison of the final cdrs) is `:unequal`.

**Comment by slburson** [comment]
That is indeed a bug.  My apologies, and thanks for the report.

**Comment by slburson** [comment]
Fixed in 7a2cb81.

**Comment by khinsen** [comment]
Thanks! That was possible the quickest bug fix I have seen happen on GitHub!
